src/main/webapp/queue.js

```diff
@@ -293,3 +293,18 @@ document.getElementById('searchArea').addEventListener('keydown', function(e) {
     getSearchQuery(document.getElementById('searchArea').value);
   }
 });
+
+// when window closes or is refreshed
+window.addEventListener('beforeunload', function(e) {
+  clearTimeouts(); // eslint-disable-line no-undef
+  if (thumbnail.style.display === 'none' && // eslint-disable-line no-undef
+      !vidOver) { // eslint-disable-line no-undef
+    removeOneViewer(); // eslint-disable-line no-undef
+  }
+  removeGuest(); // eslint-disable-line no-undef
+  deleteAnonymousUser(); // eslint-disable-line no-undef
```

Comment:
do you also mind doing the same placement for the deleteAnonymousUser function in the firebase-chat.js file for the sign out button clicked event